packages-native/intro-screen/src/IntroScreen.tsx

```diff
@@ -1,84 +1,71 @@
-import { createElement, ReactNode, useCallback, useState } from "react";
+import { createElement, useCallback, useEffect, useState } from "react";
 import { defaultWelcomeScreenStyle, IntroScreenStyle } from "./ui/Styles";
 import { IntroScreenProps } from "../typings/IntroScreenProps";
-import { Modal, Text, View, ViewStyle } from "react-native";
-import { Icon } from "mendix/components/native/Icon";
-import { DynamicValue, NativeIcon, ValueStatus } from "mendix";
-import { flattenStyles } from "@native-mobile-resources/util-widgets";
+import { Modal, View } from "react-native";
+import { DynamicValue, ValueStatus } from "mendix";
 import { SwipeableContainer } from "./SwipeableContainer";
-
-interface RenderButtonProperty {
-    [key: string]: ReactNode;
-}
+import AsyncStorage from "@react-native-community/async-storage";
+import deepmerge from "deepmerge";
 
 export function IntroScreen(props: IntroScreenProps<IntroScreenStyle>): JSX.Element {
-    const [visible, setVisible] = useState(true);
-    const styles = flattenStyles(defaultWelcomeScreenStyle, props.style);
+    const [visible, setVisible] = useState(false);
+    const customStyles = props.style ? props.style.filter(o => o != null) : [];
+    const styles =
+        customStyles && customStyles.length > 0
+            ? deepmerge.all<IntroScreenStyle>([defaultWelcomeScreenStyle, ...customStyles])
+            : defaultWelcomeScreenStyle;
+
+    useEffect(() => {
+        if (props.identifier) {
+            AsyncStorage.getItem(props.identifier).then(value => {
+                setVisible(value === "" || value === null);
+            });
+        } else {
+            setVisible(true);
+        }
+    }, []);
+
     const onDone = useCallback(() => {
         if (props.onDone && props.onDone.canExecute) {
             props.onDone.execute();
         }
-        setVisible(false);
+        hideModal();
     }, []);
+
     const onSlideChange = useCallback(() => {
         if (props.onSlideChange && props.onSlideChange.canExecute) {
             props.onSlideChange.execute();
         }
     }, []);
+
     const onSkip = useCallback(() => {
         if (props.onSkip && props.onSkip.canExecute) {
             props.onSkip.execute();
         }
-        setVisible(false);
+        hideModal();
```

Comment:
Hiding the modal before executing the onSkip action is better.

Reply:
Done